rq/worker.py

```diff
@@ -418,11 +418,22 @@ def _set_connection(self, connection: Optional['Redis']) -> 'Redis':
         """
         if connection is None:
             connection = get_current_connection()
-        current_socket_timeout = connection.connection_pool.connection_kwargs.get("socket_timeout")
-        if current_socket_timeout is None:
-            timeout_config = {"socket_timeout": self.connection_timeout}
-            connection.connection_pool.connection_kwargs.update(timeout_config)
-        return connection
+        if connection.__class__.__name__ == 'Redis':
```

Comment:
Do you mind changing this approach of explicit class name check? This runs afoul of Python's duck typing concept (e.g it wouldn't work with FakeRedis).

I'd prefer using try/except to try and detect the type of connection the worker has.

Reply:
Hi. No problem. With try excrpt will be work well.